clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -277,7 +277,7 @@ private long sendProducerData(long now) {
             }
         }
 
-        List<ProducerBatch> expiredBatches = this.accumulator.expiredBatches(this.requestTimeout, now);
+        List<ProducerBatch> expiredBatches = this.accumulator.expiredBatches(now);
```

Comment:
It seems we may release the memory for the expired batches before the response is returned. This means the underneath ByteBuffer is still referred by the ProducerBatch instance in the inFlightRequests. I am not sure if this would cause any problem, but it seems a little dangerous.